x-pack/plugins/snapshot_restore/__jest__/client_integration/snapshot_list.test.tsx

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { EuiSearchBoxProps } from '@elastic/eui/src/components/search_bar/search_box';
+
+import { useLoadSnapshots } from '../../public/application/services/http';
+import { DEFAULT_SNAPSHOT_LIST_PARAMS } from '../../public/application/lib';
+
+import * as fixtures from '../../test/fixtures';
+import { SnapshotListTestBed } from './helpers/snapshot_list.helpers';
+import { REPOSITORY_NAME } from './helpers/constant';
+import { pageHelpers, getRandomString } from './helpers';
+
+/*
+ * We are mocking useLoadSnapshots instead of sinon fake server because it's not
+ * spying on url parameters used in requests, for example /api/snapshot_restore/snapshots
+ * ?sortField=startTimeInMillis&sortDirection=desc&pageIndex=0&pageSize=20
+ * &searchField=repository&searchValue=test&searchMatch=must&searchOperator=exact
+ * would be shown as url=/api/snapshot_restore/snapshots is sinon server
+ */
+jest.mock('../../public/application/services/http', () => ({
```

Comment:
Just a note. I haven't made an "official" recommendation for it but I personally like that pattern to declare in a single place all the jest mocks for the component integration tests. It has 2 benefits:  1. reduce the size of the test file and 2. avoid declaring multiple times the same `jest.mock(...) declaration`.

In my last PR you can see

* [the single jest.mocks.ts file](https://github.com/elastic/kibana/pull/121144/files#diff-7c531153b37140a70e274b0ab370e9aa4e6e8e97f11d1719c4f943f2ab66623d)
* I import it in [a single place](https://github.com/elastic/kibana/pull/121144/files#diff-b31b498344171db73a7fdd29b72d353658967fae945685f3c7d84ae4b294c1f7R7) in `setup_environment.ts` which all the test file import
* Done. 😊